### ISSUE TITLE
Code Organization/Seperation

### ISSUE THREAD

**Issue by oshlern** [BoardGamesQuest/Board-Games#1]
### Current State

I was being hasty and bad and put the code into a file called `3x3.py`. This includes the Board and Agent classes, as well as the two sample agents RandomChoose and Human. This is bad and messy and we want to separate the code so that it is neater and easier to work on concurrently and in general.

### To Do

We should put each class in its own file (with agents in their own folder), and have a `game.py` file which imports each of the files and uses them (everything not included in the classes that is currently in `3x3.py`). This means that we define the board paramaters and compile the agents in their. We could organize it however, this is just an initial proposal cause we need to separate the code. Sorry for writing it all in one file, I was being bad and am not sure how to use code from other files in python. It would be a great learning opportunity to do this. It probably uses something like `import Board from board.py`.

**Comment by oshlern** [comment]
### Edit

The board and agents code is now in `main.py`, and there is a file called `qLearn.py` that implements it. It uses the structure suggested (`from main.py import Board`) and is functional. The code in `neural.py` isn't super important and `board.py` is out of date. There is also a `test.py` which implements `qLearn.py` and tests it. We should migrate Board code to `board.py`, Agent code to its own file, hopefully within a directory including a file for every Agent (including randomChoose). We would have to figure out all the imports needed when separating the code. Idk if this will be compatible with all the games, but there should be some abstract form that should be standard for all of our agents and boards (outside of tic tac toe). Maybe there can be an abstract subclass of Agent called ticTacAgent or something. This is more long term.

#### TL;DR
Take `main.py` and separate into `board.py` and a folder for agents. Make documents import proper stuff. Possibly put abstract classes outside `Tic Tac Toe`.

**Comment by Sora-the-explora** [comment]
Fixed Agents and Board but probably broke qLearn.
